src/bin/driver.rs

```diff
@@ -448,11 +387,21 @@ mod app {
         c.shared.network.lock(|net| net.direct_stream(target));
 
         c.shared.output_state.lock(|output| {
-            for (i, (&new_output_enabled, old_output_enabled)) in new_settings
-                .output_enabled
-                .iter()
-                .zip(old_settings.output_enabled)
-                .enumerate()
+            let (new_output_enabled, old_output_enabled) = (
+                [
+                    new_settings.output.low_noise.output_enabled,
+                    new_settings.output.high_power.output_enabled,
+                ],
+                [
+                    &old_settings.output.low_noise.output_enabled,
+                    &old_settings.output.high_power.output_enabled,
+                ],
+            );
+            for (i, (&new_output_enabled, &old_output_enabled)) in
+                new_output_enabled
+                    .iter()
+                    .zip(old_output_enabled)
+                    .enumerate()
```

Comment:
The two arrays can be in-line, `for .. in [new_settings..., new_settings....].iter().zip([old..., old...]).enumerate() {`